fix(cdk/menu): update docs to reflect current implementation and add correct role for triggers (#24884)
angular/components: src/cdk/menu/menu.md

The directives in `@angular/cdk/menu` set the appropriate roles on their host element.

| Directive           | ARIA Role        |
|---------------------|------------------|
| CdkMenuBar          | menubar          |
| CdkMenu             | menu             |
| CdkMenuGroup        | group            |
| CdkMenuItem         | menuitem         |
| CdkMenuItemRadio    | menuitemradio    |
| CdkMenuItemCheckbox | menuitemcheckbox |
| CdkMenuTrigger      | button           |

### CSS Styles and Classes

The `@angular/cdk/menu` is designed to be highly customizable to your needs. It therefore does not
make any assumptions about how elements should be styled. You are expected to apply any required
CSS styles, but the directives do apply CSS classes to make it easier for you to add custom styles.
The available CSS classes are listed below, by directive.

#### `cdkMenu`

`cdk-menu`
: Always applied
`cdk-menu-inline`
: Applied if the menu is an [inline menu](#menu-content)

#### `cdkMenuBar`

`cdk-menu-bar`
: Always applied

#### `cdkMenuGroup`

`cdk-menu-group`
: Always applied

#### `cdkMenuItem`

`cdk-menu-item`
: Always applied

#### `cdkMenuTriggerFor`

`cdk-menu-trigger`

> : Always applied

Reviewer: I'd probably make this a table like

```markdown
| Directive | CSS Class   | Applied... |
|-----------|-------------|------------|
| `CdkMenu` | `.cdk-menu` | Always     |
|           |             |            |
|           |             |            |
```

The directives in `@angular/cdk/menu` set the appropriate roles on their host element.

| Directive           | ARIA Role        |
|---------------------|------------------|
| CdkMenuBar          | menubar          |
| CdkMenu             | menu             |
| CdkMenuGroup        | group            |
| CdkMenuItem         | menuitem         |
| CdkMenuItemRadio    | menuitemradio    |
| CdkMenuItemCheckbox | menuitemcheckbox |
| CdkMenuTrigger      | button           |

### Getting started

Import the `CdkMenuModule` into the `NgModule` in which you want to create menus. You can then apply
menu directives to build your custom menu. A typical menu consists of the following directives:

- `cdkMenuTriggerFor` - links a trigger element to an `ng-template` containing the menu the trigger opens
- `cdkMenu` - creates the actual menu content that the trigger will open
- `cdkMenuItem` - added to each item in the menu

<!-- example({
  "example": "cdk-menu-standalone-trigger",
  "file": "cdk-menu-standalone-trigger-example.html"
  }) -->

Most menu interactions consist of two parts: a trigger and a menu panel.

#### Triggers

> `cdkMenuTriggerFor` can be added to any button to make it a trigger for the given menu, or any menu

Reviewer: ```suggestion
You can add `cdkMenuTriggerFor` to any button to make it a trigger for the given menu, or any menu
```
nit: passive voice

item to make it a trigger for a submenu. When adding this directive, be sure to pass a reference to

> the template containing the menu it should open. You can toggle the associated menu using a mouse

Reviewer: ```suggestion
the template containing the menu it should open. Users can toggle the associated menu using a mouse
```
I think "you" in this context is the developer writing the code vs. user who's going to use the menu

or keyboard.

<!-- example({"example":"cdk-menu-standalone-trigger",
              "file":"cdk-menu-standalone-trigger-example.html", 
              "region":"trigger"}) -->

When creating a submenu trigger, add both `cdkMenuItem` and `cdkMenuTriggerFor` like so,

<!-- example({"example":"cdk-menu-menubar",
              "file":"cdk-menu-menubar-example.html", 
              "region":"file-trigger"}) -->

#### Menu content

> Menu content is created using the `cdkMenu` or `cdkMenuBar` components. Menus can be always present

Reviewer: ```suggestion
You can create menu content by using the `cdkMenu` or `cdkMenuBar` components. Menus can be always present
```
nit: passive voice

> on the page (referred to as an inline menu), or shown on trigger activation (referred to as a popup

Reviewer: I might structure this like
```markdown
There are two types of menu:
* _inline menus_ are always present on the page
* _pop-up menus_ can be toggled to hide or show by the user
```
(also fwiw my preference is for "pop-up" vs. "popup", but both are broadly acceptable)

Author: Done, wound up having to restructure a few sections to make things flow after addressing this. May be worth another review pass over this part

menu). Popup menus must be wrapped in an `ng-template`, as a reference to the ng-template is needed
to pass to the trigger directive.

Menus should exclusively contain elements with role `menuitem`, `menuitemcheckbox`,
`menuitemradio`, or `group`. Supporting directives that automatically apply these roles are
discussed below.

Note that Angular CDK provides no styles; you must add styles as part of building your custom menu.

### Menu Bars

> The `CdkMenuBar` directive follows the [ARIA menubar][menubar] spec and behaves similar to a desktop
> app menubar. It consists of at least one `CdkMenuItem` which triggers a submenu.

Reviewer: ```suggestion
The `CdkMenuBar` directive follows the [ARIA menubar][menubar] spec and behaves similarly to a desktop
application menubar. Each bar consists of at least one `CdkMenuItem` that triggers a submenu.
```

the template containing the menu it should open. Users can toggle the associated menu using a mouse
or keyboard.

<!-- example({"example":"cdk-menu-standalone-trigger",
              "file":"cdk-menu-standalone-trigger-example.html", 
              "region":"trigger"}) -->

When creating a submenu trigger, add both `cdkMenuItem` and `cdkMenuTriggerFor` like so,

<!-- example({"example":"cdk-menu-menubar",
              "file":"cdk-menu-menubar-example.html", 
              "region":"file-trigger"}) -->

#### Menu content

There are two types of menus:
* _inline menus_ are always present on the page
* _pop-up menus_ can be toggled to hide or show by the user

You can create menus by marking their content element with the `cdkMenu` or `cdkMenuBar`
directives. You can create several types of menu interaction which are discussed below.

All type of menus should exclusively contain elements with role `menuitem`, `menuitemcheckbox`,
`menuitemradio`, or `group`. Supporting directives that automatically apply these roles are
discussed below.

Note that Angular CDK provides no styles; you must add styles as part of building your custom menu.

### Inline Menus

> An _inline menu_ is a menu that lives directly on the page rather than a pop-up associated with a

Reviewer: ```suggestion
An _inline menu_ is a menu that lives directly on the page rather than in a pop-up associated with a
```

### Menu Items

> The `cdkMenuItem` directive allows the items to be navigated to via keyboard interaction.

Reviewer: general comment: we should be consistent about "menu item" vs. "menuitem" throughout. I'd probably use "menu item" unless we're explicitly talking about the ARIA role

> The `cdkMenuItem` directive allows the items to be navigated to via keyboard interaction.
> A menuitem can provide some user defined action by hooking into the `cdkMenuItemTriggered` output.

Reviewer: ```suggestion
The `cdkMenuItem` directive allows users to navigate menuitems via keyboard.
You can add a custom action to a menuitem with the `cdkMenuItemTriggered` output.
```

<!-- example({"example":"cdk-menu-standalone-stateful-menu",
              "file":"cdk-menu-standalone-stateful-menu-example.html", 
              "region":"reset-item"}) -->

You can create nested menus by using a menuitem as the trigger for another menu.

<!-- example({"example":"cdk-menu-menubar",
              "file":"cdk-menu-menubar-example.html", 
              "region":"file-trigger"}) -->

#### Menu Item Checkboxes

A `cdkMenuItemCheckbox` is a special type of menuitem that behaves as a checkbox. You can use this

> type of menuitem to toggle items on and off. An element with  `cdkMenuItemCheckbox` directive

Reviewer: ```suggestion
type of menuitem to toggle items on and off. An element with the `cdkMenuItemCheckbox` directive
```
